src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -0,0 +1,161 @@
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.coremod.tileentities.TileEntityWareHouse;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * ----------------------- Not Documented Object ---------------------
+ */
+public class WarehouseRequestResolver extends AbstractRequestResolver<ItemStack>
+{
+
+    public WarehouseRequestResolver(
+                                     @NotNull final ILocation location,
+                                     @NotNull final IToken token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public Class<? extends ItemStack> getRequestType()
+    {
+        return ItemStack.class;
+    }
+
+    @Override
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<ItemStack> requestToCheck)
+    {
+        if (ItemStackUtils.isEmpty(requestToCheck.getRequest()))
+        {
+            return false;
+        }
+
+        final TileEntity tileEntity = manager.getColony().getWorld().getTileEntity(getLocation().getInDimensionLocation());
+
+        if (tileEntity instanceof TileEntityWareHouse)
+        {
+            final TileEntityWareHouse wareHouse = (TileEntityWareHouse) tileEntity;
+
+            return wareHouse.isInHut(requestToCheck.getRequest());
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken> attemptResolve(
+                                          @NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        final TileEntity tileEntity = manager.getColony().getWorld().getTileEntity(getLocation().getInDimensionLocation());
+
+        if (tileEntity instanceof TileEntityWareHouse)
+        {
+            final TileEntityWareHouse wareHouse = (TileEntityWareHouse) tileEntity;
+            final BlockPos pos = wareHouse.getPositionOfChestWithItemStack(request.getRequest());
+
+            request.setResult(request.getRequest().copy());
+            return Lists.newArrayList(manager.createRequest(new WarehouseChestDeliveryRequester(this, manager.getFactoryController().getNewInstance(UUID.randomUUID(),
+              new TypeToken<IToken>() {}), manager.getFactoryController().getNewInstance(pos, new TypeToken<ILocation>() {}), request.getToken()),
+              new Delivery(manager.getFactoryController().getNewInstance(
+                pos,
+                new TypeToken<ILocation>() {}), request.getRequester().getDeliveryLocation(), request.getRequest())));
+        }
+
+        return Lists.newArrayList();
+    }
+
+    @Nullable
+    @Override
+    public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request) throws RuntimeException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.RuntimeException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)